### PR TITLE
python-style loops in `sage.graphs`

### PR DESCRIPTION
**Pull request by dcoudert**
Convert the loops of the form `for x from 0 <= x < n` in `sage.graphs` to python-style loops.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/graphs/matchpoly.pyx [hunk 1]
```diff
@@ -223,7 +223,7 @@ def matching_polynomial(G, complement=True, name=None):
     if complement and G.density() > 0.5:  # this cutoff could probably be tuned
         f_comp = matching_polynomial(G.complement()).list()
         f = x.parent().zero()
-        for i from 0 <= i <= nverts / 2:  # implicit floor
+        for i in range(nverts / 2 + 1):  # implicit floor
```
Comment (fchapoton) [inline src/sage/graphs/matchpoly.pyx]: nverts // 2 + 1 maybe

Comment (dcoudert) [reply src/sage/graphs/matchpoly.pyx]: Right.

**Comment by fchapoton** [comment]
I do not understand why the checks are not running..

**Comment by dcoudert** [comment]
Me neither. I already saw that for instance in #35990.

**Comment by fchapoton** [comment]
can you please tell me if sage builds and the doctests pass in src/graphs ?

**Review by fchapoton — APPROVED** [review]
ok, merci. Feu vert

**Comment by dcoudert** [comment]
Merci à toi.

**Pull request merged**